### ISSUE TITLE
check-template-types reports incorrectly on class generics

### ISSUE THREAD

**Issue by RobertAKARobin** [gajus/eslint-plugin-jsdoc#1354]
## Expected behavior
Expected no errors.

## Actual behavior
`check-template-names` reports:
`@template U not in use`

## ESLint Config

```js
// eslint.config.mjs
import tsEslint from 'typescript-eslint';
import jsdoc from 'eslint-plugin-jsdoc';

export default [
	{
		files: [`**/*.js`],
		languageOptions: {
			parser: tsEslint.parser,
			parserOptions: {
				ecmaFeatures: {
					latest: true,
				},
				projectService: true,
			},
		},
		plugins: {
			'@typescript-eslint': tsEslint.plugin,
			jsdoc,
		},
		rules: {
			'jsdoc/check-template-names': `error`,
		},
	},
];
```

...and TSConfig, just for good measure:

```json
{
	"compilerOptions": {
		"allowJs": true,
		"checkJs": true,
		"experimentalDecorators": true,
		"lib": ["ESNext"],
		"module": "ESNext",
		"noEmit": true,
		"noImplicitOverride": true,
		"noImplicitReturns": true,
		"strict": true,
		"target": "ESNext",
		"useDefineForClassFields": false
	},
	"include": [
		"index.js"
	]
}
```

## ESLint sample

```js
// index.js
/**
 * @template {string} U // Reports error: "@template U not in use `eslint(jsdoc/check-template-names)`"
 */
export class User {
	/**
	 * @type {U}
	 */
	name;

	/**
	 * @param {U} name
	 */
	constructor(name) {
		this.name = name;
	}

	yellName() {
		return /** @type {Uppercase<U>} */(this.name.toUpperCase());
	}
}

const user = new User(`Steve`); // VSCode shows the type is correctly `User<"Steve">`
user.yellName(); // VSCode shows the type is correctly `STEVE`
```

Originally my `@template` was called `User`, so I thought this might be related to the issue of template names being more than a character long (https://github.com/gajus/eslint-plugin-jsdoc/issues/1352). But renaming the `@template` to just `U` didn't change anything.

## Environment

- Node version: 23.6.0
- ESLint version 9.19.0
- `eslint-plugin-jsdoc` version: 50.6.3

**Comment by typenoob** [comment]
To fix this, we need to find an approach to get `@param` and `@return` of the functions in the class. I don't know if there's existing api to do this.

**Comment by brettz9** [comment]
Thanks for the report. I've sought to address this in #1407 . Note that the code snippet of `return /** @type {Uppercase<U>} */(this.name.toUpperCase());` is not detected by the fix, so you would need to supply the template usage on the method's `@returns`. (It would, I think, be far too messy for us to try to examine each path for usage.)
